servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancerBuilder.java

```diff
@@ -135,13 +135,13 @@ public LoadBalancerFactory<ResolvedAddress, C> build() {
         }
         final LoadBalancerObserver<ResolvedAddress> loadBalancerObserver = this.loadBalancerObserver != null ?
                 this.loadBalancerObserver : NoopLoadBalancerObserver.instance();
-        Supplier<HealthChecker<ResolvedAddress>> healthCheckerSupplier;
+        Function<String, HealthChecker<ResolvedAddress>> healthCheckerSupplier;
```

Comment:
healthCheckerSupplier -> healthCheckerFactory?
Also, consider marking it as `final`